src/yacs-web/src/services/YacsService.js

```diff
@@ -70,14 +70,13 @@ export const getDepartments = () =>
 export const getSubSemesters = () =>
   client.get('/subsemester').then(({ data }) => {
     return data.map(subsemester => {
-      subsemester.date_start = new Date(subsemester.date_start);
-      subsemester.date_end = new Date(subsemester.date_end);
+      subsemester.date_start = localToUTCDate(new Date(subsemester.date_start));
+      subsemester.date_end = localToUTCDate(new Date(subsemester.date_end));
       subsemester.date_start_display = readableDate(subsemester.date_start);
       subsemester.date_end_display = readableDate(subsemester.date_end);
-
-      subsemester.display_string = `
-                    ${subsemester.date_start_display} - ${subsemester.date_end_display}
-                `;
+      // Used to determine what semester the subsemester is part of
+      subsemester.semester_name = subsemester.parent_semester_name;
```

Comment:
If can update `src/typedef.js` would be mch appreciated

Reply:
done